docs/website/blog/2022-06-30-binding-database-service-without-sbo.md

```diff
@@ -0,0 +1,326 @@
+---
+title: Binding to a database service without SBO
+author: Parthvi Vala
+author_url: https://github.com/valaparthvi
+author_image_url: https://github.com/valaparthvi.png
+tags: ["binding"]
+slug: binding-database-service-without-sbo
+---
+
+How to bind to your application to a database service without the Service Binding Operator.
+
+<!--truncate-->
+
+There are a few ways of binding your application to a database service with the help of odo. The recommended way is with the help of Service Binding Operator(SBO), but it is also possible to bind without it, and this blog will show you how.
+
+
+## Architecture
+We have a simple CRUD application built in Go that can create/list/update/delete a place. This application requires connecting to a MongoDB database in order to function correctly, which will be deployed as a microservice on the cluster.
+
+## Prerequisites:
+This blog assumes:
+- [odo v3.0.0-beta1](https://github.com/redhat-developer/odo/releases/tag/v3.0.0-beta1)
+- you have admin access to a Kubernetes or OpenShift cluster to be able to install the MongoDB operator.
+- you have _Helm_ installed on your system. See https://helm.sh/docs/intro/install/ for installation instructions.
+
+## (Optional) Setting up the namespace
+0. We will create a new namespace to deploy our application in, with the help of odo.
+```sh
+odo create namespace restapi-mongodb
+```
+
+## Setting up the application
+1. Clone the repository, and cd into it.
+```sh
+git clone https://github.com/valaparthvi/restapi-mongodb-odo.git && cd restapi-mongodb-odo
+```
+
+
+## Export environment variables
+2. Export the necessary environment variables:
```

Comment:
I agree, but then if we want to maintain the order of `odo init` > `odo dev` > deploy charts > wait for odo dev to detect changes, it becomes necessary to export these variables at this point

Reply:
The order 
- odo init
- odo dev
- deploy chart
is just an example provided by Tomas, I'm not sure it must be respected at all price.

By first deploying the chart (including exportint variables), then running odo init and odo dev, the flow seems more understandable to me

Reply:
> is just an example provided by Tomas, I'm not sure it must be respected at all price.
> 
> 
> By first deploying the chart (including exportint variables), then running odo init and odo dev, the flow seems more understandable to me

I like the current approach of enabling Helm repo first. But I don't agree with the idea of spinning up Helm chart using it before doing `odo init`. If it were me, I would first show that your Go application is deployed using `odo dev`, but `curl` request made to it is throwing 404 or some error. Thereafter, we spin up the MongoDB database required by the application.

Or maybe, do an `odo dev` without all the variables which would help one connect to the MongoDB database. Make a `curl` call that returns 404 or some error. Now add information about connection to MongoDB and see how the application returns expected output.

I'm saying this because that's how I think an application developer's flow would be like. But then I have not a lot of application development myself, so correct me if I'm mistaken.

Reply:
> Or maybe, do an `odo dev` without all the variables which would help one connect to the MongoDB database. Make a `curl` call that returns 404 or some error. Now add information about connection to MongoDB and see how the application returns expected output.
> 
> I'm saying this because that's how I think an application developer's flow would be like. But then I have not a lot of application development myself, so correct me if I'm mistaken.

I agree with you, but I can't find a flow that works for everyone. For now, I have added a note stating that `odo dev` will fail if run before adding connection information.

Reply:
Works for me. I discussed this with @feloy and agree with the flow we have now.